src/App.tsx

```diff
@@ -1,204 +1,123 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+
 import { UserWarning } from './UserWarning';
 import { USER_ID } from './api/todos';
+import Header from './components/Header';
+import Footer from './components/Footer';
+import TodoList from './components/TodoList';
+import { Status } from './types/Status';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { wait } from './utils/fetchClient';
 
 export const App: React.FC = () => {
+  const [todo, setTodo] = useState<string>('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<Status>(Status.All);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+
+        return wait(3000).then(() => setErrorMessage(''));
+      });
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const itemsLeft = todos.filter(({ completed }) => !completed).length;
+
+  const haveCompletedTodos = todos.some(({ completed }) => completed);
+
+  const allCompleted = todos.every(({ completed }) => completed);
+
+  const filteredTodos = todos.filter(task => {
+    switch (status) {
+      case Status.Active:
+        return !task.completed;
+      case Status.Completed:
+        return task.completed;
+      case Status.All:
+      default:
+        return true;
+    }
+  });
+
+  const addTodo = (e: React.FormEvent) => {
+    e.preventDefault();
+    if (todo.trim() !== '') {
```

Comment:
```suggestion
    if (!!todo.trim()) {
```

can be a bit simplify